cfme/utils/appliance/__init__.py

```diff
@@ -2050,6 +2050,21 @@ def update_advanced_settings(self, settings_dict):
                 raise ApplianceException('No server id is set, cannot modify yaml config via REST')
             self.server.update_advanced_settings(settings_dict)
 
+    def set_proxy(self, host, port, user=None, password=None, prov_type=None):
+        vmdb_config = self.advanced_settings
+        proxy_type = prov_type or 'default'
+        settings = {'host': host,
+                    'port': port,
+                    'user': user,
+                    'password': password}
+        try:
+            vmdb_config['http_proxy'][proxy_type] = settings
+        except KeyError as ex:
+            logger.error('Incorrect provider type')
+            logger.exception(ex)
+            raise Exception('Impossible to create proxy with current provider type')
+        return self.update_advanced_settings(vmdb_config)
```

Comment:
**QUESTION**
What should be returned here? `update_advanced_settings` returns nothing in https://github.com/ManageIQ/integration_tests/blob/master/cfme/utils/appliance/__init__.py#L2012